lblod/ember-rdfa-editor
# GN-4587: custom gap-cursor plugin

### Overview
This PR introduces a custom gap-cursor plugin containing several fixes compared to the original version:
- The `click` handler has been replaced by a `mousedown` handler in order to intercept a click event earlier
- The types of the `GapCursor` class have been fixed
- Addition of a fix when resolving the position returned by `view.posAtCoords`.

##### connected issues and PRs:
Solves the gap-cursor part of [GN-4587](https://binnenland.atlassian.net/browse/GN-4587?atlOrigin=eyJpIjoiNzQzYzZmMTA2MmE3NGY0YTlmZjM5NDRmOGQ1YzY4YzciLCJwIjoiaiJ9)

### How to test/reproduce
- Open-up the dummy app
- Insert a node with the `isolating` attribute (like a table) at the start of the document
- Notice that both in firefox and chromium-based browsers, it is now possible to click before/after the table in order to display a gap-cursor

### Challenges/uncertainties
- This PR introduce an additional function `resolvePosition` which resolves the position returned by `view.posAtCoords`. Ideally, this function is not necessary, as it tries to circumvent the issue with wrong positions returned by `view.posAtCoords`.
- The fixes included in this PR _should_ normally work in both firefox and chromium-based browsers.



### Checks PR readiness
